src/components/ArticleList.vue

```diff
@@ -96,18 +96,30 @@
     watch: {
       currentPage (newValue) {
         this.listConfig.filters.offset = (newValue - 1) * 10
-        this.$store.dispatch(FETCH_ARTICLES, this.listConfig)
+        this.fetchArticles()
+      },
+      type () {
+        this.resetPagination()
+        this.fetchArticles()
+      },
+      author () {
+        this.resetPagination()
+        this.fetchArticles()
+      },
+      tag () {
+        this.resetPagination()
+        this.fetchArticles()
+      },
+      favorited () {
+        this.resetPagination()
+        this.fetchArticles()
```

Comment:
Is it necessary to watch all the props?
Since the props are updated every time we load the component, I don't see why we should add the extra load of watching them.
Maybe you can provide an example for me to understand?

Reply:
That'w what i thougt. Then i went to my profile page and clicked on Favorited Articles and clicked on another person. I saw the profile pic but the list items were not updated. Thats becouse there was no watcher on the author so it did not reload. So i added a watcher on that one and thought why not just add a watcher to all the props, there will be a day when they change.